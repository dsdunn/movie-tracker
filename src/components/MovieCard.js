import React from 'react';
import PropTypes from 'prop-types';

export const MovieCard = ({ movie, userId, toggleFav, isFav }) => {
  const handleClick = () => {
    toggleFav(userId, movie);
  };

  return (
    <div id={movie.id} className={`movie-card ${isFav ? 'favorite' : ''}`} >
      <img 
        alt='popular movie' 
        src={`https://image.tmdb.org/t/p/w300/${movie.poster_path}`}
      />
      <h3>{movie.title}</h3>
      <h3>Rating: {movie.vote_average}</h3>
      <button onClick={handleClick}>Favorite</button>
    </div>
  );
};

MovieCard.propTypes = {
  movie: PropTypes.object,
  userId: PropTypes.number,
  toggleFav: PropTypes.func,
  isFav: PropTypes.bool
};