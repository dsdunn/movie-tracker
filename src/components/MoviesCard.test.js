import { MovieCard } from './MovieCard.js';
import React from 'react';
import { shallow } from 'enzyme';

describe('MovieCard', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const mockMovieCard = [{
      title: "Legally Blonde",
      vote_average: 100}]
    
    const wrapper = shallow(<MovieCard />)
    expect(wrapper).toMatchSnapshot()
    })
  })
