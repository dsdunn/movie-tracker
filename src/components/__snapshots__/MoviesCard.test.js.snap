// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    movie={
        Object {
            "id": 2,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="popular movie"
          src="https://image.tmdb.org/t/p/w300/undefined"
/>,
        <h3 />,
        <h3 />,
        <button
          onClick={[Function]}
>
          Favorite
</button>,
      ],
      "className": "movie-card ",
      "id": 2,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "popular movie",
          "src": "https://image.tmdb.org/t/p/w300/undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="popular movie"
            src="https://image.tmdb.org/t/p/w300/undefined"
/>,
          <h3 />,
          <h3 />,
          <button
            onClick={[Function]}
>
            Favorite
</button>,
        ],
        "className": "movie-card ",
        "id": 2,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "popular movie",
            "src": "https://image.tmdb.org/t/p/w300/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
